components/camel-lra/src/main/java/org/apache/camel/service/lra/LRASagaRoutes.java

```diff
@@ -87,12 +88,18 @@ private void verifyRequest(Exchange exchange) {
                 if (!queryParams.isEmpty()) {
                     if (queryParams.get(URL_COMPENSATION_KEY) != null) {
                         compensationURI = queryParams.get(URL_COMPENSATION_KEY).toString();
+                        // CAMEL-21216: the call from the lra-coordinator is not correctly url-decoded
+                        // as long as 'CAMEL-21197' is not solved, this workaround is needed
+                        compensationURI = java.net.URLDecoder.decode(compensationURI, StandardCharsets.UTF_8);
```

Comment:
@zhfeng 
i implemented a complete new parse method (parseQuery) in the local class to avoid a new dependency to other libs like 'org.apache.http.client.utils.URLEncodedUtils', i also set up a small junit test for it .. 

What do u think?